apps/remote_control/lib/lexical/remote_control.ex

```diff
@@ -77,10 +57,25 @@ defmodule Lexical.RemoteControl do
     :persistent_term.put({__MODULE__, :project}, project)
   end
 
+  def halt(%Project{} = project) do
+    node = node_name(project)
+    :rpc.call(node, System, :halt, [])
+  end
+
   def stop(%Project{} = project) do
-    project
-    |> node_name()
-    |> :slave.stop()
+    node = node_name(project)
+    :ok = :net_kernel.monitor_nodes(true, node_type: :visible)
+    :ok = :rpc.call(node, System, :stop, [])
+
+    receive do
+      {:nodedown, ^node, _} ->
+        :ok
+    after
+      5_000 ->
```

Comment:
Halting the project is really the responsibility of the ProjectNode genserver, though to do it correctly, you're going to need to make use of `GenServer.reply`

what should happen is that you define a function like this in the `ProjectNode` genserver:
```elixir
def stop(%Project{} = project) do 
  project
  |> name()
  |> GenServer.call(:stop)
```

and then the handle_call would look something like this:

```elixir
def handle_call(:stop, from, %State{} = state) do 
   # do the stuff to stop the node
   {:noreply, State.set_stopped_by(state, from), @stop_timeout}
end 
```
and then wait for the message and handle the timeouts.
```elixir
def handle_info({:nodedown, ^name}, state) do 
  GenServer.reply(state.stopped_by, :ok)
  {:stop, :normal, nil}
end

def handle_info(:timeout, state) do 
  # how halt the node
  {:noreply, state}
end
```